internal/utils/web.go

```diff
@@ -77,15 +77,29 @@ func (t TreeMux) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	http.NotFound(w, req)
 }
 
-const authCookieName = "renterd_auth"
+const (
+	authCookieName = "renterd_auth"
+	authQueryParam = "apikey"
+)
 
 // Auth wraps an http.Handler to force authentication with either a basic auth
 // password or a cookie.
 func Auth(password string) func(http.Handler) http.Handler {
 	return func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			if cookie, err := req.Cookie(authCookieName); err == nil && authTokens.Validate(cookie.Value) {
-				h.ServeHTTP(w, req)
+			if cookie, err := req.Cookie(authCookieName); err == nil {
+				// cookie found
+				if !authTokens.Validate(cookie.Value) {
+					httpWriteError(w, "Cookie for authentication found but is either invalid or expired", http.StatusUnauthorized)
+				} else {
+					h.ServeHTTP(w, req)
+				}
+			} else if req.URL.Query().Has(authQueryParam) {
```

Comment:
`nit` evaluate token first? check both first before sending 401?

Reply:
Changed it to do the token first to allow for overriding cookies but I'd like to avoid being too forgiving on the authentication. If you provide a token that is invalid I think you should be aware rather than maybe still passing because your browser has still got a valid cookie attached.